src/pages/home/ReportScreen.js

```diff
@@ -6,49 +6,102 @@ import styles from '../../styles/styles';
 import ReportView from './report/ReportView';
 import ScreenWrapper from '../../components/ScreenWrapper';
 import HeaderView from './HeaderView';
-import Navigation from '../../libs/Navigation/Navigation';
-import ROUTES from '../../ROUTES';
-import ONYXKEYS from '../../ONYXKEYS';
+import {fetchActions} from '../../libs/actions/Report';
 
 const propTypes = {
-    // id of the most recently viewed report
-    currentlyViewedReportID: PropTypes.string,
-};
+    /* Navigation api provided by react navigation */
+    navigation: PropTypes.shape({
+        /* Display the drawer programmatically */
+        openDrawer: PropTypes.func.isRequired,
+    }).isRequired,
 
-const defaultProps = {
-    currentlyViewedReportID: '0',
+    /* Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
+        /* Route specific parameters used on this screen */
+        params: PropTypes.shape({
+            /* The ID of the report this screen should display */
+            reportID: PropTypes.string,
+        }).isRequired,
+    }).isRequired,
 };
 
-const ReportScreen = (props) => {
-    const activeReportID = parseInt(props.currentlyViewedReportID, 10);
-    if (!activeReportID) {
-        return null;
+class ReportScreen extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+            error: null,
+        };
     }
 
-    return (
-        <ScreenWrapper
-            style={[
-                styles.appContent,
-                styles.flex1,
-                styles.flexColumn,
-            ]}
-        >
-            <HeaderView
-                reportID={activeReportID}
-                onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
-            />
-            <View style={[styles.dFlex, styles.flex1]}>
-                <ReportView reportID={activeReportID} />
-            </View>
-        </ScreenWrapper>
-    );
-};
+    componentDidMount() {
+        this.fetchReport();
+    }
+
+    componentDidUpdate(prevProps) {
+        // Reports changed, reset and load new data
+        if (this.reportID !== prevProps.route.params.reportID) {
+            this.fetchReport();
+        }
+    }
+
+    // Todo: ask why getters aren't on top?
+    get canRenderMainContent() {
+        return this.reportID && !this.state.isLoading && !this.state.error;
+    }
+
+    get reportID() {
+        const params = this.props.route.params;
+        return params.reportID;
+    }
```

Comment:
Can we use computed getters? 
Usually they will be at the top of the file, but eslint is forcing them here

Reply:
Hmm, I _think_ we should avoid this for now. While convenient, it's not really very common to see in any of our codebases.